soc/arm/nordic_nrf/common/soc_secure.h

```diff
@@ -59,7 +59,7 @@ static inline void soc_secure_gpio_pin_mcu_select(uint32_t pin_number,
 #if defined(CONFIG_SOC_HFXO_CAP_INTERNAL)
 static inline uint32_t soc_secure_read_xosc32mtrim(void)
 {
-	return NRF_FICR_S->XOSC32MTRIM;
+	return NRF_FICR->XOSC32MTRIM;
```

Comment:
is this not going to impact existing socs somehow? I'd imagine it's not but can't tell why :-)

Reply:
Good point. @magp-nordic @masz-nordic @gmarull @anangl

Reply:
> is this not going to impact existing socs somehow? I'd imagine it's not but can't tell why :-)

It is not. This is under [`!defined(CONFIG_TRUSTED_EXECUTION_NONSECURE)`](https://github.com/zephyrproject-rtos/zephyr/blob/1fcd750e922632f01db20a81b6103b8478a683d6/soc/arm/nordic_nrf/common/soc_secure.h#L43) and for existing SoCs (nRF53 and nRF91 Series) we use this translation: `#define NRF_FICR NRF_FICR_S`.